test/lint/lint-locale-dependence.py

```diff
@@ -0,0 +1,259 @@
+    "mbstowcs",     # LC_CTYPE
+    "mbtowc",       # LC_CTYPE
+    "mktime",
+    "normalize",    # boost::locale::normalize
+    "printf",       # LC_NUMERIC
+    "putwc",
+    "putwchar",
+    "scanf",        # LC_NUMERIC
+    "setlocale",
+    "snprintf",
+    "sprintf",
+    "sscanf",
+    "std::locale::global",
+    "std::to_string",
+    "stod",
+    "stof",
+    "stoi",
+    "stol",
+    "stold",
+    "stoll",
+    "stoul",
+    "stoull",
+    "strcasecmp",
+    "strcasestr",
+    "strcoll",      # LC_COLLATE
+    #"strerror",
+    "strfmon",
+    "strftime",     # LC_TIME
+    "strncasecmp",
+    "strptime",
+    "strtod",       # LC_NUMERIC
+    "strtof",
+    "strtoimax",
+    "strtol",       # LC_NUMERIC
+    "strtold",
+    "strtoll",
+    "strtoq",
+    "strtoul",      # LC_NUMERIC
+    "strtoull",
+    "strtoumax",
+    "strtouq",
+    "strxfrm",      # LC_COLLATE
+    "swprintf",
+    "to_lower",     # boost::locale::to_lower
+    "to_title",     # boost::locale::to_title
+    "to_upper",     # boost::locale::to_upper
+    "tolower",      # LC_CTYPE
+    "toupper",      # LC_CTYPE
+    "towctrans",
+    "towlower",     # LC_CTYPE
+    "towupper",     # LC_CTYPE
+    "trim",         # boost::algorithm::trim
+    "trim_left",    # boost::algorithm::trim_left
+    "trim_right",   # boost::algorithm::trim_right
+    "ungetwc",
+    "vasprintf",
+    "vdprintf",
+    "versionsort",
+    "vfprintf",
+    "vfscanf",
+    "vfwprintf",
+    "vprintf",
+    "vscanf",
+    "vsnprintf",
+    "vsprintf",
+    "vsscanf",
+    "vswprintf",
+    "vwprintf",
+    "wcrtomb",
+    "wcscasecmp",
+    "wcscoll",      # LC_COLLATE
+    "wcsftime",     # LC_TIME
+    "wcsncasecmp",
+    "wcsnrtombs",
+    "wcsrtombs",
+    "wcstod",       # LC_NUMERIC
+    "wcstof",
+    "wcstoimax",
+    "wcstol",       # LC_NUMERIC
+    "wcstold",
+    "wcstoll",
+    "wcstombs",     # LC_CTYPE
+    "wcstoul",      # LC_NUMERIC
+    "wcstoull",
+    "wcstoumax",
+    "wcswidth",
+    "wcsxfrm",      # LC_COLLATE
+    "wctob",
+    "wctomb",       # LC_CTYPE
+    "wctrans",
+    "wctype",
+    "wcwidth",
+    "wprintf"
+]
+
+
+def git_grep_call(regexp_locale_dependent_functions):
+    exclude_args = [":(exclude)" + excl for excl in REGEXP_EXTERNAL_DEPENDENCIES_EXCLUSIONS]
+    git_grep_command = ["git", "grep", "-E", "[^a-zA-Z0-9_\\`'\"<>](" +  regexp_locale_dependent_functions + "(_r|_s)?)[^a-zA-Z0-9_\\`'\"<>]", "--", "*.cpp", "*.h"] + exclude_args
+    git_grep_output = list()
+
+    try:
+        git_grep_output = check_output(git_grep_command, universal_newlines=True, encoding="utf8").splitlines()
+    except CalledProcessError as e:
+        if e.returncode > 1:
+            raise e
+
+    return git_grep_output
+
+
+def main():
+    exit_code = 0
+
+    regexp_ignore_known_violations = "|".join(KNOWN_VIOLATIONS)
+    regexp_locale_dependent_functions = "|".join(LOCALE_DEPENDENT_FUNCTIONS)
+    git_grep_output = git_grep_call(regexp_locale_dependent_functions)
```

Comment:
I think it'd make sense to pass in the list of locale dependent functions to this function, then build the regexp inside (it's a implementation detail).